tfs/collection.py

```diff
@@ -180,82 +249,75 @@ def __init__(self, parent, attr):
             self.attr = attr
 
         def __getitem__(self, plane: str):
-            return getattr(self.parent, self.attr + "_" + plane.lower())
+            return getattr(self.parent, f"{self.attr}_{plane.lower()}")
 
         def __setitem__(self, plane: str, value):
-            setattr(self.parent, self.attr + "_" + plane.lower(), value)
+            setattr(self.parent, f"{self.attr}_{plane.lower()}", value)
+
+    class _FilenameGetter:
+        def __init__(self, parent: 'TfsCollection'):
+            self.parent = parent
+
+        def __getitem__(self, item) -> str:
+            return self.parent.get_filename(item)
+
+        def __getattr__(self, attr) -> str:
+            return self[attr]
+
+        def __call__(self, exist: bool = False) -> Dict[str, str]:
+            all_filenames = {name: self.parent.get_filename(name)
+                             for name in self.parent.defined_properties}
+            if not exist:
+                return all_filenames
+            return {name: filename for name, filename in all_filenames.items()
+                    if (self.parent.directory / filename).exists()}
```

Comment:
Should this call `.is_file()` instead? It could be tricked by the (unlikely) case where something is present with this name, but not a file.